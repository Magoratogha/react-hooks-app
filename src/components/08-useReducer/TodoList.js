import PropTypes from 'prop-types';
import { memo } from 'react';
import TodoListItem from './TodoListItem';

const TodoList = memo(({ todos, handleToggle, handleDelete }) => {
    return (
        <ul className="list-group list-group-flush">
            {
                todos.map((todo, index) =>
                    <TodoListItem
                        key={ todo.id }
                        todo={ todo }
                        index={ index }
                        handleToggle={ handleToggle }
                        handleDelete={ handleDelete }/>)
            }
        </ul>
    );
});

TodoList.propTypes = {
    todos: PropTypes.array.isRequired,
    handleToggle: PropTypes.func.isRequired,
    handleDelete: PropTypes.func.isRequired
};

export default TodoList;