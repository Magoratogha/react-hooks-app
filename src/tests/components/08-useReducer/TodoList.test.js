import { queryAllByAttribute, render } from "@testing-library/react";
import TodoList from "../../../components/08-useReducer/TodoList";
import { todosMock } from "../../mocks/todos";

describe('Test on <TodoList />', () => {
    
    const handleToggleMock = jest.fn();
    const handleDeleteMock = jest.fn();

    test('should render properly', () => {
        const { container } = render(<TodoList 
            todos={ todosMock }
            handleToggle={ handleToggleMock }
            handleDelete={ handleDeleteMock } />);
        expect(container).toMatchSnapshot();
    });

    test('should contains 2 todo items', () => {
        const { container } = render(<TodoList 
            todos={ todosMock }
            handleToggle={ handleToggleMock }
            handleDelete={ handleDeleteMock } />);
        const todoItems = queryAllByAttribute('class', container, 'list-group-item', { exact: true });
        expect(todoItems.length).toBe(todosMock.length);
    });
});