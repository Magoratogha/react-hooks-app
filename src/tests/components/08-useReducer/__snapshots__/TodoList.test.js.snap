// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <TodoList /> should render properly 1`] = `
<div>
  <ul
    class="list-group list-group-flush"
  >
    <li
      class="list-group-item"
    >
      <p
        class="mb-0 "
      >
        1
        . 
        To learn React
      </p>
      <button
        class="btn btn-outline-danger btn-sm"
      >
        Delete
      </button>
    </li>
    <li
      class="list-group-item"
    >
      <p
        class="mb-0 "
      >
        2
        . 
        To learn Mongo
      </p>
      <button
        class="btn btn-outline-danger btn-sm"
      >
        Delete
      </button>
    </li>
  </ul>
</div>
`;
